ktor-server/ktor-server-plugins/ktor-server-openapi/jvm/src/io/ktor/server/plugins/openapi/OpenAPI.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.plugins.openapi
+
+import io.ktor.server.http.content.*
+import io.ktor.server.routing.*
+import io.swagger.codegen.v3.*
+import io.swagger.codegen.v3.generators.html.*
+import java.io.*
+
+/**
+ * Creates a `get` endpoint at [path] with documentation rendered from the OpenAPI file at [filename].
+ *
+ * The documentation is generated using [StaticHtml2Codegen] by default. It can be customized using config in [block].
+ * See [OpenAPIConfig] for more details.
+ */
+public fun Routing.openApi(path: String, filename: String, block: OpenAPIConfig.() -> Unit = {}) {
```

Comment:
same as with swagger integration - filename/File/ByteReadChannel? (no nix support here of course)

Reply:
Please support serving from resources